src/parser/shared/modules/items/shadowlands/crafted/DarkmoonDeckVoracity.tsx

```diff
@@ -0,0 +1,214 @@
+      category: Abilities.SPELL_CATEGORIES.ITEMS,
+      cooldown: 90,
+      gcd: null,
+      castEfficiency: {
+        suggestion: true,
+        recommendedEfficiency: 0.8,
+      },
+    });
+
+    (options.buffs as Buffs).add({
+      spellId: SPELLS.VORACIOUS_HASTE.id,
+      timelineHighlight: true,
+    });
+
+    (options.statTracker as StatTracker).add(SPELLS.VORACIOUS_HASTE.id, {
+      haste: this.lastCard.hasteDrain,
+    });
+
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(DARKMOON_DECK_VORACITY_CARDS), this.removeCardBuff);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.VORACIOUS_HASTE), this.applyHasteBuff);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.VORACIOUS_HASTE), this.removeHasteBuff);
+  }
+
+  removeCardBuff(event: RemoveBuffEvent) {
+    this.lastCardId = event.ability.guid;
+    this.lastCard = DARKMOON_DECK_VORACITY_CARDS_INFO[this.lastCardId];
+  }
+
+  applyHasteBuff(event: ApplyBuffEvent) {
+    this.statTracker.update(SPELLS.VORACIOUS_HASTE.id, {
+      haste: this.lastCard.hasteDrain,
+    });
+    if (!this.lastCardId || !DARKMOON_DECK_VORACITY_CARDS_INFO[this.lastCardId]) {
+      return;
+    }
+    DARKMOON_DECK_VORACITY_CARDS_INFO[this.lastCardId].hastePercent += DARKMOON_DECK_VORACITY_CARDS_INFO[this.lastCardId].hasteDrain / this.statTracker.ratingNeededForNextPercentage(this.statTracker.currentHasteRating, this.statTracker.statBaselineRatingPerPercent[STAT.HASTE]);
+    this.lastUseTimestamp = event.timestamp;
+  }
+
+  removeHasteBuff(event: RemoveBuffEvent) {
+    if (!this.lastCardId || !DARKMOON_DECK_VORACITY_CARDS_INFO[this.lastCardId]) {
+      return;
+    }
+    DARKMOON_DECK_VORACITY_CARDS_INFO[this.lastCardId].uptime = event.timestamp - this.lastUseTimestamp;
+  }
+
+  getTotals(deck: Array<{ name: string; hasteDrain: number; hastePercent: number; uptime: number; }>) {
+    let totalUptime = 0;
+    let totalPercent = 0;
+    Object.values(deck).forEach((card) => {
+      totalUptime += card.uptime;
+      totalPercent += card.hastePercent;
+    });
+    return {
+      totalUptime: totalUptime,
+      totalPercent: totalPercent,
+    };
+  }
+
+  getAverageHastePercentage(hastePercentage: number) {
+    return hastePercentage * this.selectedCombatant.getBuffUptime(SPELLS.VORACIOUS_HASTE.id) / this.owner.fightDuration;
+  }
+
+  statistic() {
+    const filtered = Object.values(DARKMOON_DECK_VORACITY_CARDS_INFO).filter(function(card) {
+      if (card.uptime > 0) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+    const totals = this.getTotals(filtered);
+    return (
+      <Statistic
+        position={STATISTIC_ORDER.CORE()}
+        size="flexible"
+        category={STATISTIC_CATEGORY.ITEMS}
+        dropdown={
+          <table className="table table-condensed">
+            <thead>
+              <tr>
+                <th>Name</th>
```

Comment:
I feel like name is weird here. Wouldn't card be a better name

Reply:
Sure